test/aqua.jl

```diff
@@ -0,0 +1,36 @@
+using Test
+using StatsBase
+using Aqua
+
+@testset "Aqua tests (performance)" begin
+    # This tests that we don't accidentally run into
+    # https://github.com/JuliaLang/julia/issues/29393
+    # Aqua.test_unbound_args(StatsBase)
+    ua = Aqua.detect_unbound_args_recursively(StatsBase)
+    @test length(ua) == 0
+
+    # See: https://github.com/SciML/OrdinaryDiffEq.jl/issues/1750
+    # Test that we're not introducing method ambiguities across deps
+    ambs = Aqua.detect_ambiguities(StatsBase; recursive = true)
+    # Uncomment for debugging:
+    # for method_ambiguity in ambs
+    #     @show method_ambiguity
+    # end
+    @test length(ambs) ≤ 5
+    potentially_pirated_methods = Aqua.Piracy.hunt(StatsBase)
+    # Uncomment for debugging:
+    # for pirate in potentially_pirated_methods
+    #     @show pirate
+    # end
+    @test length(potentially_pirated_methods) ≤ 27
+end
+
+@testset "Aqua tests (additional)" begin
+    Aqua.test_undefined_exports(StatsBase)
```

Comment:
Just run `Aqua.test_all` and disable the ones you're handling separately? Then we won't miss out on new tests potentially added in the future.

Reply:
While I do like the idea of not missing out on new tests in the future, my issue with `test_all` is that, if something does fail, then the simplest fix is often to call the individual test functions.